lib/validator/validator.go

```diff
@@ -364,6 +364,26 @@ func validateRule(v *validator.Validate, structLevel *validator.StructLevel) {
 				"Destination.NotPorts", "", reason("protocol does not support ports"))
 		}
 	}
+
+	// Check for mismatch in IP versions
+	if rule.Source.Net != nil && rule.IPVersion != nil {
+		if rule.Source.Net.Version() != *(rule.IPVersion) {
+			structLevel.ReportError(reflect.ValueOf(rule.Source.Net), "Source.Net",
+				"", reason("rule does not support mixing of IPv4/v6 CIDRs"))
```

Comment:
I think we can say that the rule contains an IP version match that does not match the source/dest CIDR version.